src/component/Tooltip.tsx

```diff
@@ -202,101 +196,120 @@ export class Tooltip<TValue extends ValueType, TName extends NameType> extends P
+    [`${CLS_PREFIX}-bottom`]:
+      isNumber(translateY) && coordinate && isNumber(coordinate.y) && translateY >= coordinate.y,
+    [`${CLS_PREFIX}-top`]: isNumber(translateY) && coordinate && isNumber(coordinate.y) && translateY < coordinate.y,
+  });
 
-    const cls = classNames(CLS_PREFIX, {
-      [`${CLS_PREFIX}-right`]:
-        isNumber(translateX) && coordinate && isNumber(coordinate.x) && translateX >= coordinate.x,
-      [`${CLS_PREFIX}-left`]: isNumber(translateX) && coordinate && isNumber(coordinate.x) && translateX < coordinate.x,
-      [`${CLS_PREFIX}-bottom`]:
-        isNumber(translateY) && coordinate && isNumber(coordinate.y) && translateY >= coordinate.y,
-      [`${CLS_PREFIX}-top`]: isNumber(translateY) && coordinate && isNumber(coordinate.y) && translateY < coordinate.y,
-    });
+  return (
+    // ESLint is disabled to allow listening to the `Escape` key. Refer to
+    // https://github.com/recharts/recharts/pull/2925
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+    <div
+      role="dialog"
+      onKeyDown={event => {
+        if (event.key === 'Escape') {
+          setState(prev => ({
+            ...prev,
+            dismissed: true,
+            dismissedAtCoordinate: {
+              ...prev.dismissedAtCoordinate,
+              x: coordinate.x,
+              y: coordinate.y,
+            },
+          }));
+        }
+      }}
+      className={cls}
+      style={outerStyle}
+      ref={wrapperNode}
+    >
+      {renderContent(content, {
+        ...props,
+        payload: finalPayload,
+      })}
+    </div>
+  );
+};
 
-    return (
-      // ESLint is disabled to allow listening to the `Escape` key. Refer to
-      // https://github.com/recharts/recharts/pull/2925
-      // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
-      <div
-        tabIndex={-1}
-        role="dialog"
-        onKeyDown={event => {
-          if (event.key === 'Escape') {
-            this.setState({
-              dismissed: true,
-              dismissedAtCoordinate: {
-                ...this.state.dismissedAtCoordinate,
-                x: this.props.coordinate.x,
-                y: this.props.coordinate.y,
-              },
-            });
-          }
-        }}
-        className={cls}
-        style={outerStyle}
-        ref={node => {
-          this.wrapperNode = node;
-        }}
-      >
-        {renderContent(content, { ...this.props, payload: finalPayload })}
-      </div>
-    );
-  }
-}
+// needs to be set so that renderByOrder can find the correct handler function
+Tooltip.displayName = 'Tooltip';
+
+/**
+ * needs to be set so that renderByOrder can access an have default values for
+ * children.props when there are no props set by the consumer
+ * doesn't work if using default parameters
+ */
+Tooltip.defaultProps = tooltipDefaultProps;
+
+export { Tooltip };
```

Comment:
Nit: Why export it in an object? Why not simply `export Tooltip`?

Reply:
oh, habit I guess -  I'm used to exporting multiple things at the bottom of a file